Importing necessary libraries

In [1]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
from os import path

Defining file paths and configurations

In [2]:
DATA_DIR = './data'
TRAIN_FILE = 'train.csv'
TEST_FILE = 'test.csv'

In [3]:
train_df = pd.read_csv(path.join(DATA_DIR, TRAIN_FILE), index_col='PassengerId')
train_df.head()

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

In [4]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 891 entries, 1 to 891
Data columns (total 11 columns):
Survived    891 non-null int64
Pclass      891 non-null int64
Name        891 non-null object
Sex         891 non-null object
Age         714 non-null float64
SibSp       891 non-null int64
Parch       891 non-null int64
Ticket      891 non-null object
Fare        891 non-null float64
Cabin       204 non-null object
Embarked    889 non-null object
dtypes: float64(2), int64(4), object(5)
memory usage: 83.5+ KB


We see that there are 891 entries in the train dataset. 

There seem to be fewer entries in the **Age**, **Cabin** and **Embarked** columns. Let us visualize these columns to understand how we can preprocess these features.

**Age Column**

Since about 20% of the values are missing in the Age column, we will impute these values using the following logic:
1. Split the Title from Name column.
2. Study Age distribution based on Title.
3. Replace missing ages with mean age based on Title.

The derived Title column can be related to both the Sex and the Age columns, since Sex can be inferred from the Title, we will not explicitly use the Sex column for determining Age.